### PR TITLE
gh-115058: Add ``reset_rare_event_counters`` function in `_testinternalcapi`

### PR DESCRIPTION
**Pull request by Eclips4**
* Issue: gh-115058

### CODE REVIEW

**Comment by Eclips4** [comment]
Does it makes sense to call `_testinternalcapi.reset_rare_event_counters` in every test case in this module?

**Review by mdboom — APPROVED** [review]
Thanks.  LGTM!

File: Lib/test/test_optimizer.py [hunk 1]
```diff
@@ -43,6 +43,7 @@ def test_set_eval_frame_func(self):
         _testinternalcapi.set_eval_frame_default()
 
     def test_builtin_dict(self):
+        _testinternalcapi.reset_rare_event_counters()
```
Comment (sobolevn) [inline Lib/test/test_optimizer.py]: There are multiple other places where `get_rare_event_counters` is used. Why only a sinlge place handles reset?

Comment (Eclips4) [reply Lib/test/test_optimizer.py]: Yep, I'm talking about the same in https://github.com/python/cpython/pull/115128#issuecomment-1936252960.
I'm going to add a reset in other places.

File: Modules/_testinternalcapi.c [hunk 2]
```diff
@@ -1650,6 +1650,20 @@ get_rare_event_counters(PyObject *self, PyObject *type)
     );
 }
 
+static PyObject *
+reset_rare_event_counters(PyObject *self, PyObject *type)
```
Comment (markshannon) [inline Modules/_testinternalcapi.c]: `type` is unused, can you rename it to `unused` or similar?

Comment (Eclips4) [reply Modules/_testinternalcapi.c]: I marked it as ``Py_UNUSED``

**Pull request merged**